### ISSUE TITLE
poly check does not check test dependencies

### ISSUE THREAD

**Issue by seancorfield** [polyfy/polylith#272]
As discussed on Slack. @tengstrand asked me to create this issue.

**Describe the bug**
`poly check` does a great job of making sure the project `:deps` contain all the necessary bricks but if the `:aliases` > `:test` > `:extra-deps` add more bricks, `poly check` does not check whether those bricks' transitive dependencies are satisfied -- this can lead to a very painful process of doing `poly test`, getting a ns not found error, adding that brick, running `poly test` again and getting a different ns not found error (missing transitive brick), rinse and repeat... I think poly check should also check test dependencies and indicate which of those are missing separately from the main deps (I know it's a harder problem to solve but it would have saved me from quite a bit of pain, several times, during our big migration).

**To Reproduce**
1. Have some tests in a project depend on a brick A that is not a regular runtime dep for that project and which has a transitive dependency on brick B, also not a regular runtime dep for that project, 
2. Run `poly check` -- it will say `OK`,
3. Run `poly test` -- it will fail because brick B is missing from the `:test` alias,
4. Add brick B to the project's `:test` alias and run `poly test` again -- it will pass.

**Expected behavior**
`poly check` should analyze `:test` dependencies and report missing bricks (separately from `:deps` bricks).

**Comment by tengstrand** [comment]
I found the bug. Now it should work. Please have a look.

**Comment by seancorfield** [comment]
When I first ran it, it said `affiliate` was a missing component in several projects' test deps. I was puzzled because all our tests passed -- but then I realized I had `affiliate` as a test dep directly in one of the components that those projects used. I moved that test dep from the component's test deps to each of the projects' test deps that it identified and it was satisfied. Good.

However, we have an odd case in one of our bases, where some test setup depends on code that is in another base. The two bases are closely related and pulling this code apart into a component, just to address the base test issue would be both complicated and not really worth the effort. So we have that test dep in one base, referring to the other base. I commented out this test dep -- and `poly check` did not report that it was missing (but the tests failed -- because the dep _is_ needed).

I'm not sure whether this is a case that `poly check` should catch or not? I understand the argument for pulling apart the base, so that both bases can refer to a new component -- one as a regular dep, the other as a test dep -- but it truly isn't worth the pain (it's the core (Sierra) Component for the base and drags in nearly all of that application so I'd pretty much have to make that entire base into a component just to satisfy this test dep). We're okay having this one, known, "exception" to `poly check`'s rules.

**Comment by tengstrand** [comment]
Taken from a conversation with Sean:

And the bases involved are `billing-rebill` whose tests depend on `billing-member`.

Specifically `ws.billing-rebill.chargeback911-test` and `ws.billing-rebill.rebill-test` both depend on `ws.billing-member.web` which is the main ns of `billing-member` and contains the primary "system" Component (Sierra) construction function -- which chargeback and rebill tests both need to construct and drive some member-facing activity in order to test certain backend functionality

That Sierra Component in turn depends on almost every part of the billing-member app (base) -- controllers, model, etc -- so splitting it out into a component is both a lot of work and, more importantly, unnecessary for anything except these two test nses.

**Comment by tengstrand** [comment]
Please have a look @seancorfield and see if the [issue-293](https://github.com/polyfy/polylith/tree/issue-293) branch solves the remaining problem.

**Comment by seancorfield** [comment]
Current setup, with `billing-member` base dependency in `billing-rebill` base `:test` deps:
```
wsmain$ check
  Error 107: Missing components in the wsbilling-rebill project, for the test context, for these interfaces: billing-member
```
Move `billing-member` base dependency to `wsbilling-rebill` project `:test` deps:
```
wsmain$ check
OK
```
Excellent!

Thank you!

**Comment by tengstrand** [comment]
Okay, nice that it worked! Another change is that the `deps` command now shows dependencies to bases.

**Comment by seancorfield** [comment]
Unfortunately, a side-effect of this "fix" is that now `poly test` thinks the entire `billing-member` base has to be retested in the context of the `wsbilling-rebill` project since it is a (test) dependency. Previously, it just "didn't notice" the dependency existed and did not attempt to test both bases together, just the `billing-rebill` base. And that breaks things in all sorts of interesting ways for us, let alone the big increase in test run time. So we can't actually use this version of Polylith -- even tho' it is "doing the right thing" -- unless we tackle some version of the refactoring we were trying to avoid by doing this in the first place... 😞

**Comment by tengstrand** [comment]
Hi. I think that will be solved by issue #274 which I will soon start working on. I will merge #293 to master and continue with issue 274.
